src/rust/engine/src/intrinsics.rs

```diff
@@ -641,8 +644,13 @@ fn interactive_process(
       .await?;
 
     let code = exit_status.code().unwrap_or(-1);
-    if keep_sandboxes == KeepSandboxes::OnFailure && code != 0 {
+    if keep_sandboxes == KeepSandboxes::Always
+        || keep_sandboxes == KeepSandboxes::OnFailure && code != 0 {
       tempdir.keep("interactive process");
+      let cwd = current_dir()
```

Comment:
I couldn't find an obvious way to call current_dir() inside the `if run_in_workpace` branch, since the Path cannot outlive the PathBuf it's created from. But I assume there has to be some idiomatic way?

Reply:
One way is to move the function call that needs the reference from below the conditional to inside the conditional. To dry that up, extract the function call as a lambda above the conditional:
```
let lambda = |x| {...};
if cond { let y = ...; lambda(&y) } else { lamda(&z) }
```

Then you only allocate in the if branch but retain the scope you need.

Reply:
Makes sense, done.